ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala

```diff
@@ -70,34 +76,33 @@ private[h2] object PseudoHeaders {
     val path = findWithNoDuplicates(pseudos)(_._1 === PATH).map(_._2)
     val authority = extractAuthority(pseudos)
 
-    val noOtherPseudo = pseudos
-      .filterNot(t => requestPsedo.contains(t._1))
-      .isEmpty
+    val noOtherPseudo = pseudos.forall(t => requestPsedo.contains(t._1))
 
-    val teIsCorrect = headers.find(_._1 === "te").headOption.fold(true)(_._2 === "trailers")
-    val connectionIsAbsent = headers.find(_._1 === "connection").isEmpty
+    val teIsCorrectAndNoConnection = !headers.exists { case (k, v) =>
```

Comment:
`!exists(_ == foo)` is equal to `forall(_ != foo)`, isn't?

Reply:
Yeah... it's one of this cases in which one may want to think this in English... Which one would be easier to say?

> All headers must not be neither 1. a connection, nor 2. a "te" without "trailers" 

> No header can be either 1. a connection, or 2. a "te" without "trailers"

The former bends easier to `.forall( case (k, v) => ! (---) && ! (---) ) `, whereas the latter look closer to the code as is.